### ISSUE TITLE
Can't chain indexers

### ISSUE THREAD

**Issue by psybers** [boalang/compiler#74]
Code like the following:

````
m: map[string] of map[int] of bool;
m["s"][0] = true;
````

does not work.  This gives a codegen error, because it will generate something like `____m.put("s").get(0, true);` due to the workaround for put() we have in place currently.

**Comment by psybers** [comment]
A simple work around for this bug is the change `m["s"][0] = true` into a sequence of statements:

````
m2 := m["s"];
m2[0] = true;
````

**Comment by psybers** [comment]
Fixed by aea3f23a87b6b52d2cc57e31b8ec71cc379b820f.
